core/includes/date.class.inc

```diff
@@ -9,6 +9,22 @@
 class BackdropDateTime extends DateTime {
   public $granularity = array();
   public $errors = array();
+
+  /**
+   * @var bool
+   */
+  public $dateOnly;
+
+  /**
+   * @var bool
+   */
+  public $timeOnly;
+
+  /**
+   * @var string
+   */
+  public $originalTime;
+
```

Comment:
:wink: as the code change is big per se, I took care to not touch anything not directly related to necessary changes for our task.

No, they don't need defaults here, as those get set in the constructor (as before). We only have to define the properties, this class will use.

Reply:
Fair enough again. The the only problem is inconsistency. Perhaps a follow-up issue to sort that out, but I thought that since we are making these changes here now, might as well do things properly.

Reply:
More consistency for BackdropDateTime should definitely be a follow-up. That might need more consideration and discussion, unrelated to PHP 8.2 compatibility.

But... good hint, I just realized, that $time isn't always a string, the doc block should reflect that, of course. Thanks! :+1: